api/v1/intrusiondetection_types.go

```diff
@@ -36,6 +36,27 @@ type IntrusionDetectionSpec struct {
 	// IntrusionDetectionControllerDeployment configures the IntrusionDetection Controller Deployment.
 	// +optional
 	IntrusionDetectionControllerDeployment *IntrusionDetectionControllerDeployment `json:"intrusionDetectionControllerDeployment,omitempty"`
+
+	// DeepPacketInspectionDaemonset configures the DPI Daemonset
+	// +optional
+	DeepPacketInspectionDaemonset *DeepPacketInspectionDaemonset `json:"deepPacketInspectionDaemonset,omitempty"`
+}
+
+type DeepPacketInspectionDaemonset struct {
```

Comment:
The pattern we want to use here is to create a struct that mimics the behavior and structure of the `v1.DaemonSet` object in upstream kubernetes (with only the fields that we want to expose).

So, we want to include the full resource hierarchy here. Take a look at [IntrusionDetectionControllerDeployment](https://github.com/tigera/operator/blob/dcb21dfb148f47b5dfd39bd136398e04c4c21935/api/v1/intrusiondetection_types.go#L100-L105) for an example.

In this case, something like this:

```go
type DeepPacketInspectionDaemonset struct {
  Spec *DPIDaemonsetSpec
}

type DPIDaemonsetSpec struct {
  Template *DPIDaemonsetTemplate
}

type DPIDaemonsetTemplate struct {
  Spec *DPIDaemonsetTemplateSpec
}

type DPIDaemonsetTemplateSpec struct {
  InitContainers []DPIDaemonsetInitContainer
}

type DPIDaemonsetInitContainer struct {
  Image string
  Resources *corev1.ResourceRequirements
}
```

this is more verbose than what you have written, so might seem like overkill, but allows us to extend in the future if we want to allow further configuration of the DaemonSet (e.g., metadata.labels, metadata.annotations, or other spec fields) and is identitcal to the structure of a built-in k8s resource that users are already very familiar with.

Reply:
That's now corrected.